Src/IronPython.Modules/resource.cs

```diff
@@ -0,0 +1,427 @@
+        => RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ?
+            (int)macos__rlimit_resource.RLIMIT_FSIZE : (int)linux__rlimit_resource.RLIMIT_FSIZE;
+
+    public static int RLIMIT_DATA
+        => RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ?
+            (int)macos__rlimit_resource.RLIMIT_DATA : (int)linux__rlimit_resource.RLIMIT_DATA;
+
+    public static int RLIMIT_STACK
+        => RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ?
+            (int)macos__rlimit_resource.RLIMIT_STACK : (int)linux__rlimit_resource.RLIMIT_STACK;
+
+    public static int RLIMIT_CORE
+        => RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ?
+            (int)macos__rlimit_resource.RLIMIT_CORE : (int)linux__rlimit_resource.RLIMIT_CORE;
+
+    public static int RLIMIT_RSS
+        => RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ?
+            (int)macos__rlimit_resource.RLIMIT_RSS : (int)linux__rlimit_resource.RLIMIT_RSS;
+
+    public static int RLIMIT_AS
+        => RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ?
+            (int)macos__rlimit_resource.RLIMIT_AS : (int)linux__rlimit_resource.RLIMIT_AS;
+
+    public static int RLIMIT_MEMLOCK
+        => RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ?
+            (int)macos__rlimit_resource.RLIMIT_MEMLOCK : (int)linux__rlimit_resource.RLIMIT_MEMLOCK;
+
+    public static int RLIMIT_NPROC
+        => RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ?
+            (int)macos__rlimit_resource.RLIMIT_NPROC : (int)linux__rlimit_resource.RLIMIT_NPROC;
+
+    public static int RLIMIT_NOFILE
+        => RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ?
+            (int)macos__rlimit_resource.RLIMIT_NOFILE : (int)linux__rlimit_resource.RLIMIT_NOFILE;
+
+    [PythonHidden(PlatformID.MacOSX)]
+    public static int RLIMIT_OFILE => RLIMIT_NOFILE;
+
+    [PythonHidden(PlatformID.MacOSX)]
+    public static int RLIMIT_LOCKS => (int)linux__rlimit_resource.RLIMIT_LOCKS;
+
+    [PythonHidden(PlatformID.MacOSX)]
+    public static int RLIMIT_SIGPENDING => (int)linux__rlimit_resource.RLIMIT_SIGPENDING;
+
+    [PythonHidden(PlatformID.MacOSX)]
+    public static int RLIMIT_MSGQUEUE => (int)linux__rlimit_resource.RLIMIT_MSGQUEUE;
+
+    [PythonHidden(PlatformID.MacOSX)]
+    public static int RLIMIT_NICE => (int)linux__rlimit_resource.RLIMIT_NICE;
+
+    [PythonHidden(PlatformID.MacOSX)]
+    public static int RLIMIT_RTPRIO => (int)linux__rlimit_resource.RLIMIT_RTPRIO;
+
+    [PythonHidden(PlatformID.MacOSX)]
+    public static int RLIMIT_RTTIME => (int)linux__rlimit_resource.RLIMIT_RTTIME;
+
+    private static int RLIM_NLIMITS
+        => RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ?
+            (int)(macos__rlimit_resource.RLIM_NLIMITS) : (int)(linux__rlimit_resource.RLIM_NLIMITS);
+
+    public static int RUSAGE_SELF => 0;
+
+    public static int RUSAGE_CHILDREN => -1;
+
+    [PythonHidden(PlatformID.MacOSX)]
+    public static int RUSAGE_THREAD => 1;
+
+    #endregion
+
+    [LightThrowing]
```

Comment:
Any reason why chose to go `LightThrowing` throughout? Is it common to hit a throwing codepath?

Reply:
The code I saw was trying to increase the soft limit of number of open files, and ignoring the error if it failed. I suspect in many cases increasing limits is an optional optimization, so the try-ignore pattern may be common.

I was debating with myself a little whether I should only raise `OSError` as a light exception, or also `ValueError`. While there is a legitimate use case for light-ignoring an `OSError here, a `ValueError` (usually) means that the caller has made an erroneous call and perhaps the call should fail loud and clear. But I don't expect this module to be used a lot from the C# side; I wrote it primarily because other packages depend on it (directly or indirectly). And given that the functions can already return a light exception and the confusing usage of `ValueError` by `resource` in CPython, I decided to return a light exception in all cases, to keep the door open for full CPython compatibility, if it ever becomes a need. But I could definitely reconsider this choice.